### PR TITLE
`--chdir=DIR` in `env` (fixes #1568)

### LINKED ISSUE #1568: `--chdir=DIR` in `env`
Latest GNU coreutils have a useful (to me) feature of changing the "current" directory before running the command:

```
  -C, --chdir=DIR      change working directory to DIR
```
The way it works in GNU coreutils is that it changes to the fully-symlink-resolved path (think `readlink -f`) of the directory being provided. I personally prefer it to be "change to the exact path I'm providing, regardless of whether it is relative or not", but maybe someone has a different opinion about this.

### PR DESCRIPTION
**Pull request by FelipeLema**
`--chdir=DIR` in `env` (fixes #1568)

### CODE REVIEW

**Comment by sylvestre** [comment]
Could you please add a test for this?
thanks

File: tests/by-util/test_env.rs [hunk 1]
```diff
@@ -150,3 +157,60 @@ fn test_fail_null_with_program() {
     let out = new_ucmd!().arg("--null").arg("cd").fails().stderr;
     assert!(out.contains("cannot specify --null (-0) with command"));
 }
+
+#[cfg(not(windows))]
```
Comment (sylvestre) [inline tests/by-util/test_env.rs:161]: Could you please fix the fmt issue? thanks

Comment (FelipeLema) [reply tests/by-util/test_env.rs:161]: manually ran `rustfmt` on this file. Should look good now.

**Comment by sylvestre** [comment]
thanks!

**Pull request merged**